test_pyproxy_gc_destroy failure in Node.js

The `test_pyproxy_gc_destroy;`  test consistently fails in Node.js (including on main) with,
```
    @pytest.mark.skip_pyproxy_check
    def test_pyproxy_gc_destroy(selenium):
...
>       assert dict(a) == {
            0: 2,
            1: 3,
            2: 4,
            3: 4,
            4: 2,
            "destructor_ran": True,
        }
E       AssertionError: assert {0: 2, 1: 3, 2: 4, 3: 4, ...} == {0: 2, 1: 3, 2: 4, 3: 4, ...}
E         Omitting 4 identical items, use -vv to show
E         Differing items:
E         {4: 3} != {4: 2}
E         Right contains 1 more item:
E         {'destructor_ran': True}
E         Full diff:
E         - {0: 2, 1: 3, 2: 4, 3: 4, 4: 2, 'destructor_ran': True}
E         + {0: 2, 1: 3, 2: 4, 3: 4, 4: 3}

src/tests/test_pyproxy.py:667: AssertionError
```

Introduced after https://github.com/pyodide/pyodide/pull/1870 was merged.

--- a/src/core/pyproxy.js
+++ b/src/core/pyproxy.js
@@ -31,6 +31,8 @@ Module.isPyProxy = isPyProxy;
 
 if (globalThis.FinalizationRegistry) {
   Module.finalizationRegistry = new FinalizationRegistry(([ptr, cache]) => {
+    cache.leaked = true;
+    pyproxy_decref_cache(cache);
     try {
       Module._Py_DecRef(ptr);
     } catch (e) {
@@ -201,10 +203,10 @@ function pyproxy_decref_cache(cache) {
   if (cache.refcnt === 0) {
     let cache_map = Module.hiwire.pop_value(cache.cacheId);
     for (let proxy_id of cache_map.values()) {
-      Module.pyproxy_destroy(
-        Module.hiwire.pop_value(proxy_id),
-        pyproxy_cache_destroyed_msg
-      );
+      const cache_entry = Module.hiwire.pop_value(proxy_id);
+      if (!cache.leaked) {
+        Module.pyproxy_destroy(cache_entry, pyproxy_cache_destroyed_msg);
+      }
     }
   }
 }

--- a/src/tests/test_pyproxy.py
+++ b/src/tests/test_pyproxy.py
@@ -657,6 +657,7 @@ def test_pyproxy_gc_destroy(selenium):
         """
     )
     selenium.collect_garbage()
+    selenium.collect_garbage()
     selenium.run(
         """
         get_ref_count(4)

